### PR TITLE
[Console] Escape synopsis output

### LINKED ISSUE #41315: [Console] Command usage output invalid if argument name is a reserved style
**Symfony version(s) affected**: >=4.4

**Description**
If you name an console input argument "info" and try to use the command without arguments, the help output is incorrect.

**How to reproduce**
```php
class FooBarCommand extends Command
{
    protected static $defaultName = 'foo:bar';

    protected function configure()
    {
        $this->addArgument('first', InputArgument::REQUIRED);
        $this->addArgument('info', InputArgument::REQUIRED);
        $this->addArgument('last', InputArgument::REQUIRED);
    }

    protected function execute(InputInterface $input, OutputInterface $output)
    {
        return Command::SUCCESS;
    }
}
```

Command line:
```
php bin\console foo:bar
[Not anough arguments info]
foo:bar <first> <last>
```

**Possible Solution**
Do not print the arguments as tags, use another delimiter instead or the output should not get formatted.

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #41315 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

**Comment by chalasr** [comment]
@jschaedl Thanks for the PR. 
Can you change the PR base branch to 4.4?

**Comment by jschaedl** [comment]
Hi @chalasr 

> Can you change the PR base branch to 4.4?

Sure. It was actually meant for and based on 4.4. 🙈

File: src/Symfony/Component/Console/Tests/Command/CommandTest.php [hunk 1]
```diff
@@ -192,7 +192,8 @@ public function testGetSynopsis()
         $command = new \TestCommand();
         $command->addOption('foo');
         $command->addArgument('bar');
-        $this->assertEquals('namespace:name [--foo] [--] [<bar>]', $command->getSynopsis(), '->getSynopsis() returns the synopsis');
+        $command->addArgument('info');
+        $this->assertEquals('namespace:name [--foo] [--] [<bar> [<info>]]', $command->getSynopsis(), '->getSynopsis() returns the synopsis');
```
Comment (jschaedl) [inline src/Symfony/Component/Console/Tests/Command/CommandTest.php:196]: This was already working before the fixed output escaping in `finishRenderThrowableOrException()`. I just added it to make sure the problem is not located in `getSynopsis()`.

File: src/Symfony/Component/Console/Application.php [hunk 2]
```diff
@@ -833,7 +833,7 @@ public function renderThrowable(\Throwable $e, OutputInterface $output): void
     private function finishRenderThrowableOrException(OutputInterface $output): void
     {
         if (null !== $this->runningCommand) {
-            $output->writeln(sprintf('<info>%s</info>', sprintf($this->runningCommand->getSynopsis(), $this->getName())), OutputInterface::VERBOSITY_QUIET);
+            $output->writeln(sprintf('<info>%s</info>', OutputFormatter::escape(sprintf($this->runningCommand->getSynopsis(), $this->getName()))), OutputInterface::VERBOSITY_QUIET);
```
Comment (jschaedl) [inline src/Symfony/Component/Console/Application.php:836]: Not sure how I can add a test case for that. @chalasr Do you have an idea?

Comment (chalasr) [reply src/Symfony/Component/Console/Application.php:836]: I think you can borrow one from ApplicationTest e.g. https://github.com/symfony/symfony/blob/14540c1ae54806e1c3b0b23934776ce3733d76b1/src/Symfony/Component/Console/Tests/ApplicationTest.php#L853-L866

Comment (chalasr) [reply src/Symfony/Component/Console/Application.php:836]: (let me know if you are stuck at reproducing this specific case :))

Comment (jschaedl) [reply src/Symfony/Component/Console/Application.php:836]: Thanks! What do you think? 6ee41c658ab9a7175ce8bf9469fc7b085db5bf22

File: src/Symfony/Component/Console/Tests/ApplicationTest.php [hunk 3]
```diff
@@ -868,9 +868,12 @@ public function testRenderExceptionEscapesLines()
...
         putenv('COLUMNS=22');
-        $application->register('foo')->setCode(function () {
-            throw new \Exception('dont break here <info>!</info>');
-        });
+        $application
+            ->register('foo')
+            ->setCode(function () {
+                throw new \Exception('dont break here <info>!</info>');
+            })
+        ;
```
Comment (chalasr) [inline src/Symfony/Component/Console/Tests/ApplicationTest.php]: Please revert the CS change to keep the diff as small as possible :)

**Review by chalasr — COMMENTED** [review]
One minor comment

File: src/Symfony/Component/Console/Tests/ApplicationTest.php (hunk 3 above)
Comment (jschaedl) [reply src/Symfony/Component/Console/Tests/ApplicationTest.php]: done

**Review by chalasr — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @jschaedl.

**Pull request merged**